provider/resources.go

```diff
@@ -432,17 +427,23 @@ func preConfigureCallbackWithLogger(credentialsValidationRun *atomic.Bool, gcpCl
 			}),
 		}
 
-		// validate the gcloud config
-		err := config.LoadAndValidate(ctx)
-		if err != nil {
-			return fmt.Errorf(noCredentialsErr, err)
-		}
-
 		skipRegionValidation := tfbridge.ConfigBoolValue(
 			vars, "skipRegionValidation", []string{"PULUMI_GCP_SKIP_REGION_VALIDATION"},
 		)
 
-		if !skipRegionValidation && config.Region != "" && config.Project != "" {
+		disableGlobalProjectWarning := tfbridge.ConfigBoolValue(
+			vars, "disableGlobalProjectWarning", []string{"PULUMI_GCP_DISABLE_GLOBAL_PROJECT_WARNING"},
+		)
+
+		if !disableGlobalProjectWarning {
+			// validate the gcloud config
+			err := config.LoadAndValidate(ctx)
+			if err != nil {
+				return fmt.Errorf(noCredentialsErr, err)
+			}
+		}
+
+		if !skipRegionValidation && disableGlobalProjectWarning && config.Region != "" && config.Project != "" {
```

Comment:
This looks wrong. Did you mean `! disableGlobalProjectWarning`.

I think an early return on line 437 would be more readable:

```golang
if disableGlobalProjectWarning {
    return
}
```